### PR TITLE
[TASK] Overhaul Create new pagetype

### PR DESCRIPTION
**Pull request by linawolf**
* correct indentation
* move code to includes
* copy code from EXT:examples

Remove duplicated information and too many info boxes

### CODE REVIEW

File: Documentation/ApiOverview/PageTypes/CreateNewPageType.rst [hunk 1]
```diff
@@ -1,208 +1,82 @@
...
 
-.. versionchanged:: 12.0
-   Starting with 12.0 a new PageDoktypeRegistry was introduced replacing the
-   :php:`$GLOBALS['PAGES_TYPES']` array.
-
-.. note::
-    In versions below TYPO3 v12.0 the creation was done in the
-    :file:`ext_tables.php` file, please use the version selector to look up
+..  versionchanged:: 12.0
+    Starting with 12.0 a new PageDoktypeRegistry was introduced replacing the
```
Comment (brotkrueml) [inline Documentation/ApiOverview/PageTypes/CreateNewPageType.rst:16]:
```suggestion
    A new :php:`PageDoktypeRegistry` was introduced replacing the
```

The "Starting with 12.0" is redundant to the versionchanged directive.

File: Documentation/ApiOverview/PageTypes/CreateNewPageType.rst [hunk 2]
```diff
@@ -1,208 +1,82 @@
...
+    :php:`$GLOBALS['PAGES_TYPES']` array. Use the version selector to look up
     the syntax in the corresponding documentation version.
```
Comment (brotkrueml) [inline Documentation/ApiOverview/PageTypes/CreateNewPageType.rst:18]:
```suggestion
    the syntax in the corresponding documentation version for older TYPO3 versions.
```

File: Documentation/ApiOverview/PageTypes/CreateNewPageType.rst [hunk 3]
```diff
@@ -1,208 +1,82 @@
...
 
 The first step is to add the new page type to the PageDoktypeRegistry described above. Then you need to add
```
Comment (brotkrueml) [inline Documentation/ApiOverview/PageTypes/CreateNewPageType.rst:20]:
```suggestion
The first step is to add the new page type to the :php:`\TYPO3\CMS\Core\DataHandling\PageDoktypeRegistry`. Then you need to add
```

"Above" it is not described.

File: Documentation/ApiOverview/PageTypes/CreateNewPageType.rst [hunk 4]
```diff
@@ -1,208 +1,82 @@
...
 the icon chosen for the new page type and allow users to drag and drop the new page type to the page
 tree.
 
-.. note::
-
-   You have to change :code:`example` in the argument of the anonymous function
-   to your own extension key.
-
 The new page type is added to the :php:`PageDoktypeRegistry::class` in
```
Comment (brotkrueml) [inline Documentation/ApiOverview/PageTypes/CreateNewPageType.rst:24]:
```suggestion
The new page type is added to the :php:`PageDoktypeRegistry` in
```

File: Documentation/ApiOverview/PageTypes/CreateNewPageType.rst [hunk 5]
```diff
@@ -1,208 +1,82 @@
...
-              'type' => 'web',
-              'allowedTables' => '*',
-          ]
-      );
+:file:`ext_tables.php`:
 
-   })();
+..  literalinclude:: _ext_tables.php
+    :language: php
+    :caption: EXT:example/ext_tables.php
```
Comment (brotkrueml) [inline Documentation/ApiOverview/PageTypes/CreateNewPageType.rst:29]:
```suggestion
    :caption: EXT:examples/ext_tables.php
```

File: Documentation/ApiOverview/PageTypes/CreateNewPageType.rst [hunk 6]
```diff
@@ -1,208 +1,82 @@
...
-* Page is site-root (`<doktype>-root`)
+*   Page contains content from another page (`<doktype>-contentFromPid`)
+*   Page is hidden in navigation (`<doktype>-hideinmenu`)
+*   Page is site-root (`<doktype>-root`)
 
-.. note::
+..  note::
 
-   Make sure to add the additional icons using the icon registry!
+    Make sure to add the additional icons using the icon registry!
```
Comment (brotkrueml) [inline Documentation/ApiOverview/PageTypes/CreateNewPageType.rst:70]:
```suggestion
    Make sure to add the additional icons using the :ref:`Icon API <icon>`!
```

File: Documentation/ApiOverview/PageTypes/CreateNewPageType.rst [hunk 7]
```diff
@@ -1,208 +1,82 @@
...
 
 
 Further Information
 -------------------
 
-.. rst-class:: compact-list
+..  rst-class:: compact-list
```
Comment (brotkrueml) [inline Documentation/ApiOverview/PageTypes/CreateNewPageType.rst:76]: Do we need those?

**Comment by brotkrueml** [comment]
I assume this should also be merged into 12.4?

File: Documentation/ApiOverview/PageTypes/CreateNewPageType.rst [hunk 8]
```diff
@@ -1,208 +1,82 @@
...
 ====================
 Create new Page Type
 ====================
 
 The following example adds a new page type called "Archive".
 
-.. include:: /Images/AutomaticScreenshots/PageTypes/NewPageType.rst.txt
+..  include:: /Images/AutomaticScreenshots/PageTypes/NewPageType.rst.txt
 
 The whole code to add a page type is shown below with the according file names above.
```
[the remark this answers is not in the document]
Comment (linawolf) [reply Documentation/ApiOverview/PageTypes/CreateNewPageType.rst]: changed the sentence

File: Documentation/ApiOverview/PageTypes/CreateNewPageType.rst [hunk 9]
```diff
@@ -1,208 +1,82 @@
...
+    :language: php
+    :caption: EXT:example/ext_tables.php
 
-User TSconfig is added in :file:`ext_localconf.php`:
+We need to add the following user tsconfig
+to all users so that the new page type is displayed in the wizard:
 
-.. code-block:: php
-   :caption: EXT:example/ext_localconf.php
+..  literalinclude:: _UserConfiguration.tsconfig
```
[the remark this answers is not in the document]
Comment (linawolf) [reply Documentation/ApiOverview/PageTypes/CreateNewPageType.rst:36]: thanks for catching this

File: Documentation/ApiOverview/PageTypes/CreateNewPageType.rst [hunk 10]
```diff
@@ -1,208 +1,82 @@
...
-   })();
+As you can see from the example, to make sure you get the correct icons, you can utilize :php:`typeicon_classes`.
 
 For the following cases you need to configure icons explicitly, otherwise they will automatically fall back to the
 variant for regular page doktypes.
 
-* Page contains content from another page (`<doktype>-contentFromPid`)
-* Page is hidden in navigation (`<doktype>-hideinmenu`)
-* Page is site-root (`<doktype>-root`)
+*   Page contains content from another page (`<doktype>-contentFromPid`)
```
[the remark this answers is not in the document]
Comment (linawolf) [reply Documentation/ApiOverview/PageTypes/CreateNewPageType.rst]: added some examples, checked in the core, the first key is diffferen

File: Documentation/ApiOverview/PageTypes/_pages.php [hunk 11]
```diff
@@ -0,0 +1,16 @@
+<?php
+
+defined('TYPO3') or die();
+
+// SAME as registered in ext_tables.php
+$customPageDoktype = 116;
```
[the remark this answers is not in the document]
Comment (linawolf) [reply Documentation/ApiOverview/PageTypes/_pages.php]: done

File: Documentation/ApiOverview/PageTypes/CreateNewPageType.rst (hunk 7 above)
Comment (linawolf) [reply Documentation/ApiOverview/PageTypes/CreateNewPageType.rst]: not really

**Comment by linawolf** [comment]
@brotkrueml please have another look

**Review by brotkrueml — APPROVED** [review]

**Pull request merged**